File: app/src/main/java/me/devsaki/hentoid/activities/sources/FapalityActivity.java
package me.devsaki.hentoid.activities.sources;

import me.devsaki.hentoid.enums.Site;

public class FapalityActivity extends BaseWebActivity {

    private static final String DOMAIN_FILTER = "fapality.com";
    private static final String[] GALLERY_FILTER = {"fapality.com/.*/[0-9]+/.*/$"};
    private static final String[] DIRTY_ELEMENTS = {".had"};

    Site getStartSite() {
        return Site.FAPALITY;
    }

    @Override
    boolean allowMixedContent() {
        return false;
    }


    @Override
    protected CustomWebViewClient getWebClient() {
        addDirtyElements(DIRTY_ELEMENTS);
        CustomWebViewClient client = new CustomWebViewClient(GALLERY_FILTER, this);
        client.restrictTo(DOMAIN_FILTER);
        return client;
    }
}

File: app/src/main/java/me/devsaki/hentoid/activities/sources/FapalityActivity.kt
package me.devsaki.hentoid.activities.sources

import me.devsaki.hentoid.enums.Site

private const val DOMAIN_FILTER = "fapality.com"
private val GALLERY_FILTER = arrayOf("fapality.com/.*/[0-9]+/.*/\$")
private val DIRTY_ELEMENTS = arrayOf(".had\",\"a[rel^='nofollow noopener']")

class FapalityActivity : BaseBrowserActivity() {
    override fun getStartSite(): Site {
        return Site.FAPALITY
    }

    override fun allowMixedContent(): Boolean {
        return false
    }


    override fun createWebClient(): CustomWebViewClient {
        val client = CustomWebViewClient(getStartSite(), GALLERY_FILTER, this)
        client.restrictTo(DOMAIN_FILTER)
        client.addRemovableElements(*DIRTY_ELEMENTS)
        return client
    }
}